### PR TITLE
[LowerToHW] Lower i0 operand in printf-encoded assertions to i1

### PR DESCRIPTION
**Pull request by uenoku**
This supports i0 value operand of verification statements. Following the implementation of printf, we use i1 value as a result.

### CODE REVIEW

File: test/Conversion/FIRRTLToHW/lower-to-hw.mlir [hunk 1]
```diff
@@ -518,17 +518,19 @@ firrtl.circuit "Simple"   attributes {annotations = [{class =
...
+    // CHECK-NEXT: [[FALSE:%.+]] = hw.constant false
     // CHECK-NEXT: [[TRUE:%.+]] = hw.constant true
     // CHECK-NEXT: [[TMP1:%.+]] = comb.xor %enable, [[TRUE]]
     // CHECK-NEXT: [[TMP2:%.+]] = comb.or bin [[TMP1]], %cond
     // CHECK-NEXT: sv.assert.concurrent posedge %clock, [[TMP2]] message "assert0"
     // CHECK-NEXT: sv.ifdef "USE_PROPERTY_AS_CONSTRAINT" {
     // CHECK-NEXT:   sv.assume.concurrent posedge %clock, [[TMP2]]
     // CHECK-NEXT: }
-    firrtl.assert %clock, %cond, %enable, "assert1"(%value) : !firrtl.uint<42> {isConcurrent = true, format = "ifElseFatal"}
+    firrtl.assert %clock, %cond, %enable, "assert1"(%value, %i0) : !firrtl.uint<42>, !firrtl.uint<0> {isConcurrent = true, format = "ifElseFatal"}
```
Comment (dtzSiFive) [inline test/Conversion/FIRRTLToHW/lower-to-hw.mlir:533]: Nit: Can this string be changed so "substitutions" are used?  I know we don't inspect it presently, but as-is it doesn't seem entirely unreasonable to elide the substitutions as trivially unused.

**Review by dtzSiFive — APPROVED** [review]
LGTM.

This seems reasonable for `%d` format specifier in particular, and for many others not sure we should worry about what "right semantics" are here.  Aligning with printf is nice!

File: test/Conversion/FIRRTLToHW/lower-to-hw.mlir (hunk 1 above)
Comment (uenoku) [reply test/Conversion/FIRRTLToHW/lower-to-hw.mlir]: Nice, thanks!

**Review by darthscsi — APPROVED** [review]
Seems reasonable.

**Pull request merged**